### PR TITLE
feat: remove discard action modal

### PR DESCRIPTION
**Pull request by el**
## Changes

@rebelchris @sshanzel Please test this thoroughly. I have updated https://preview.app.daily.dev/ to point to this branch for easy testing.

### CODE REVIEW

File: packages/shared/src/components/modals/NewCommentModal.tsx [hunk 1]
```diff
@@ -78,15 +78,26 @@ export default function NewCommentModal({
...
+      title: 'Discard comment',
+      description: 'Are you sure you want to close and discard your comment?',
+      okButton: {
+        className: 'btn-primary-ketchup',
+        title: 'Discard',
+      },
+      cancelButton: {
+        title: 'Stay',
+      },
+    };
```
Comment (sshanzel) [inline packages/shared/src/components/modals/NewCommentModal.tsx:92]: I think we can take this outside of the component and be declared just a const.

File: packages/shared/src/components/modals/OnboardingModal.tsx [hunk 2]
```diff
@@ -70,8 +70,23 @@ function OnboardingModal({
...
+      description: 'If you leave your changes will not be saved',
+      okButton: {
+        className: 'btn-primary-ketchup',
+        title: 'Discard',
+      },
+      cancelButton: {
+        title: 'Stay',
+        className: 'btn-primary-cabbage',
+      },
+    };
```
Comment (sshanzel) [inline packages/shared/src/components/modals/OnboardingModal.tsx:84]: Tagging for visibility.

File: packages/shared/src/hooks/useAuthForms.ts [hunk 3]
```diff
@@ -20,31 +19,45 @@ interface UseAuthFormsProps {
...
+        className: 'btn-secondary',
+      },
+      cancelButton: {
+        title: 'Stay',
+        className: 'btn-primary-cabbage',
+      },
+      className: {
+        buttons: 'flex-row-reverse',
+      },
+    };
```
Comment (sshanzel) [inline packages/shared/src/hooks/useAuthForms.ts:40]: Tagging for visibility.

File: packages/webapp/pages/signup.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ function Signup(): ReactElement {
   const { formRef } = useAuthForms();
   const router = useRouter();
 
-  const onClose = () => {
+  const onClose = async (): Promise<void> => {
```
Comment (sshanzel) [inline packages/webapp/pages/signup.tsx:14]: I think we can revert this change as there seem to be no `await` call was added

**Review by sshanzel — COMMENTED** [review]
I have reviewed code-wise (will test the preview deployment after this) - everything looks amazing and it is so much cleaner now. Thank you for removing the numerous generic discard modals. Really a sight to see 🙏 

I just had one general comment with regards to the prompt options. I think we can those outside the component itself. Probably just a regular const inside the file.

**Comment by sshanzel** [comment]
Not sure they should be both primary buttons.
image


Everything else, I have checked works as intended! Great work! 🚢

File: packages/shared/src/components/modals/OnboardingModal.tsx (hunk 2 above)
Comment (el) [reply packages/shared/src/components/modals/OnboardingModal.tsx]: Fixed, thank you 🙏

File: packages/shared/src/hooks/useAuthForms.ts [hunk 5]
```diff
@@ -20,31 +19,45 @@ interface UseAuthFormsProps {
...
     if (forceClose || !formRef?.current) {
       return onDiscard(e, forceClose);
     }
-    return setIsDiscardOpen(true);
-  };
-
-  const onDiscardCanceled: CloseAuthModalFunc = (e) => {
-    e.stopPropagation();
-    setIsDiscardOpen(false);
+    return openPrompt(e);
```
Comment (el) [inline packages/shared/src/hooks/useAuthForms.ts:50]: @sshanzel I dont think this prompt will be shown anywhere. The only place that uses `onDiscardAttempt` function is already passing forceClose=true in AuthOptions.tsx: 
```
  const onProfileSuccess = async () => {
    await refetchBoot();
    onSuccessfulRegistration?.();
    onClose(null, true);
  };
```

Can we clear up this or am I missing something?

Comment (sshanzel) [reply packages/shared/src/hooks/useAuthForms.ts:50]: I believe this is used in AuthModal.

Comment (rebelchris) [reply packages/shared/src/hooks/useAuthForms.ts:50]: Yeah it's a edge case for the callback on the authForm

**Review by rebelchris — COMMENTED** [review]
Think Lee covered all my points already.
Nice cleanup 🙏

File: packages/shared/src/components/modals/NewCommentModal.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/modals/NewCommentModal.tsx]: Makes sense. Done.

File: packages/shared/src/hooks/useAuthForms.ts (hunk 5 above)
Comment (el) [reply packages/shared/src/hooks/useAuthForms.ts:51]: Thanks, should be fine now.

File: packages/webapp/pages/signup.tsx (hunk 4 above)
Comment (el) [reply packages/webapp/pages/signup.tsx]: Fixed ❤️

File: packages/shared/src/components/modals/OnboardingModal.tsx [hunk 6]
```diff
@@ -70,8 +70,26 @@ function OnboardingModal({
     if (forceClose) {
       return onRequestClose(e);
     }
-
-    return setIsClosing(true);
+    const options: PromptOptions = {
```
Comment (rebelchris) [inline packages/shared/src/components/modals/OnboardingModal.tsx:73]: These can also go outside the component.

**Review by rebelchris — CHANGES_REQUESTED** [review]
One more to go 🥳

**Review by sshanzel — APPROVED** [review]
Your work on the generic prompt made this cleanup possible. Can't say nothing but great work! ✨ ✨ ✨

File: packages/shared/src/components/modals/OnboardingModal.tsx (hunk 6 above)
Comment (el) [reply packages/shared/src/components/modals/OnboardingModal.tsx]: Done

**Review by rebelchris — APPROVED** [review]
Awesome!
Just make sure to double check once it hits production 🙏
